eva-server/src/main/java/uk/ac/ebi/eva/server/ws/VariantWSServerV2.java

```diff
@@ -0,0 +1,227 @@
+public class VariantWSServerV2 extends EvaWSServer {
+
+    @Autowired
+    private VariantWithSamplesAndAnnotationsService service;
+
+    @GetMapping(value = "/{variantId}")
+    public Resource<QueryResponse> getCoreInfo(@PathVariable("variantId") String variantId,
+                                               @RequestParam(name = "species") String species,
+                                               @RequestParam(name = "annot-vep-version", required = false)
+                                                       String annotationVepVersion,
+                                               @RequestParam(name = "annot-vep-cache-version", required = false) String
+                                                       annotationVepCacheVersion,
+                                               HttpServletResponse response) {
+        initializeQuery();
+
+        String errorMessage = checkErrorHelper(variantId, annotationVepVersion, annotationVepCacheVersion, species);
+        if (errorMessage != null) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return new Resource<>(setErrorQueryResponse(errorMessage));
+        }
+
+        MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName(species));
+
+        List<VariantWithSamplesAndAnnotation> variantEntities;
+        Long numTotalResults;
+
+        try {
+            variantEntities = getVariantEntitiesByParams(variantId, annotationVepVersion, annotationVepCacheVersion);
+            numTotalResults = (long) variantEntities.size();
+        } catch (AnnotationMetadataNotFoundException ex) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return new Resource<>(setQueryResponse(ex.getMessage()));
+        }
+
+        List<VariantWithSamplesAndAnnotation> rootVariantEntities = new ArrayList<>();
+        variantEntities.forEach(variantEntity -> {
+            VariantWithSamplesAndAnnotation variant = new VariantWithSamplesAndAnnotation(variantEntity.getChromosome(),
+                    variantEntity.getStart(), variantEntity.getEnd(), variantEntity.getReference(),
+                    variantEntity.getReference(), variantEntity.getMainId());
+
+            variantEntity.getIds().forEach(id -> variant.addId(id));
+            rootVariantEntities.add(variant);
+        });
+
+        QueryResult<VariantWithSamplesAndAnnotation> queryResult = buildQueryResult(rootVariantEntities,
+                numTotalResults);
+
+        Link annotationLink = new Link(linkTo(methodOn(VariantWSServerV2.class).getAnnotations(variantId, species,
+                annotationVepVersion, annotationVepCacheVersion, response)).toUri().toString(), "annotation");
+
+        Link sourceEntriesLink = new Link(linkTo(methodOn(VariantWSServerV2.class).getSourceEntries(variantId, species,
+                annotationVepVersion, annotationVepCacheVersion, response)).toUri().toString(), "sourceEntries");
```

Comment:
Shall we rename public references as "sources" instead of using the weirder "sourceEntries"?

Reply:
Okay, I will change it

Reply:
Fixed :+1: